cli/keys.go

```diff
@@ -3,66 +3,125 @@
 package cli
 
 import (
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/elliptic"
 	"crypto/rand"
 	"crypto/rsa"
 	"crypto/x509"
 	"encoding/pem"
 	"log"
 	"os"
+	"reflect"
 
 	"github.com/spf13/cobra"
 )
 
 const (
 	keyBitSize     = 4096
-	privateKeyType = "RSA PRIVATE KEY"
+	rsaKeyType     = "PRIVATE KEY"
+	ecdsaKeyType   = "EC PRIVATE KEY"
+	ed25519KeyType = "PRIVATE KEY"
 	publicKeyType  = "PUBLIC KEY"
 	publicKeyFile  = "public.pem"
 	privateKeyFile = "private.pem"
 )
 
 func (cli *CLI) NewKeysCmd() *cobra.Command {
 	return &cobra.Command{
-		Use:   "keys",
+		Use:   "keys <algorithm>",
```

Comment:
make algorithm be passed by a flag and be optional. if the flag is not set, generate using rsa